src/main/java/org/junitpioneer/jupiter/resource/ResourceExtension.java

```diff
@@ -0,0 +1,368 @@
+	}
+
+	private static final class UncheckedParameterResolutionException extends RuntimeException {
+
+		public UncheckedParameterResolutionException(ParameterResolutionException cause) {
+			super(cause);
+		}
+
+		@Override
+		public synchronized ParameterResolutionException getCause() {
+			return (ParameterResolutionException) super.getCause();
+		}
+
+	}
+
+	private static String testMethodDescription(ExtensionContext extensionContext) {
+		return extensionContext.getTestMethod().map(method -> "method [" + method + ']').orElse("an unknown method");
+	}
+
+	@Override
+	public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+			ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public <T> T interceptTestFactoryMethod(Invocation<T> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		return runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public void interceptDynamicTest(Invocation<Void> invocation, DynamicTestInvocationContext invocationContext,
+			ExtensionContext extensionContext) throws Throwable {
+		runDynamicTestSequentially(invocation, extensionContext);
+	}
+
+	@Override
+	public void interceptTestTemplateMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public <T> T interceptTestClassConstructor(Invocation<T> invocation,
+			ReflectiveInvocationContext<Constructor<T>> invocationContext, ExtensionContext extensionContext)
+			throws Throwable {
+		return runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public void interceptBeforeAllMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public void interceptAfterAllMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public void interceptBeforeEachMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
+
+	@Override
+	public void interceptAfterEachMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		runSequentially(invocation, invocationContext, extensionContext);
+	}
```

Comment:
@nipafx I wasn't able to find a way to test these methods, so `ResourcesParallelismTests` doesn't cover them. Do you have any strong feelings about this?

Reply:
Wouldn't it be enough to add lifecycle methods to some of the tests to see these methods executed?

Reply:
@nipafx Sorry for the delay! I found some time this evening to start properly reading through your messages and do more coding. (I expect to have more time this coming Friday.)

Oh, good observation! I've just pushed a new commit that addresses this: [`2e8a377`](https://github.com/junit-pioneer/junit-pioneer/tree/2e8a3770428881729a207f5b42a69bffd6c9d5d2).

Reply:
Yesterday I noticed that we actually have a process for updating JUnit 5 minor versions - see #594.

Reply:
So when this is ready to be merged, we quickly create a PR for #594, merge it, then update this PR and merge it.

Reply:
@nipafx Good to know that we have a process on upgrading JUnit 5 versions, thanks! I'll keep this thread unresolved until #594 is done. :)

Reply:
Jupiter dependency was updated. ✅